app/src/org/commcare/android/util/SigningUtil.java

```diff
@@ -0,0 +1,220 @@
+     * directly because the conversion from Base64 can have a non-1:1 correspondence with the actual
+     * bytes
+     *
+     * @param textMessage
+     * @return the binary representation of the signtature
+     * @throws Exception if any errors are encountered
+     */
+    public static byte[] getSignatureBytes(String textMessage) throws Exception {
+        byte[] messageBytes = getBytesFromSMS(textMessage);
+        int lastSpaceIndex = getSignatureStartIndex(messageBytes);
+        int signatureByteLength = messageBytes.length - lastSpaceIndex;
+        byte[] signatureBytes = new byte[signatureByteLength];
+        for(int i = 0; i< signatureByteLength; i++){
+            signatureBytes[i] = messageBytes[i + lastSpaceIndex];
+        }
+        return signatureBytes;
+    }
+
+    public static int getLastSpaceIndex(String textMessage) throws Exception{
+        byte[] messageBytes = getBytesFromSMS(textMessage);
+        return getSignatureStartIndex(messageBytes);
+    }
+
+    /**
+     * Iterate through the byte array until we find the third "space" character (represented
+     * by integer 32) and then return its index
+     * @param messageBytes the raw bytes of the Base64 message
+     * @return index of the third "space" byte, -1 if none encountered
+     */
+    private static int getSignatureStartIndex(byte[] messageBytes) {
+        int index = 0;
+        int spaceCount = 0;
+        int spaceByte = 32;
+        for(byte b: messageBytes){
+            if(b == spaceByte){
+                if(spaceCount == 2){
+                    return index + 1;
+                }
+                else{
+                    spaceCount++;
+                }
+            }
+            index++;
+        }
+        return -1;
+    }
+
+    /**
+     * @param textMessage the plain text SMS message
+     * @return the parsed out profile link
+     */
+    public static String getDownloadLink(String textMessage){
+        String downloadLink  = textMessage.substring(textMessage.indexOf("ccapp: ") + "ccapp: ".length(),
+                textMessage.indexOf("signature") - 1);
+        return downloadLink;
+    }
+
+    // given a text message, parse out the cruft and return the raw Base64 bytes
+    public static byte[] getBytesFromSMS(String newMessage) throws Exception{
+        return Base64.decode(parseMessage(newMessage));
+    }
+    // given a text message, parse out the cruft and return the raw String
+    private static String parseAndDecodeSMS(String newMessage) throws Exception {
+        String parsed = parseMessage(newMessage);
+        int lastSpaceIndex = getSignatureStartIndex(getBytesFromSMS(newMessage));
+        String messageString = parsed.substring(0, lastSpaceIndex);
+        return decodeEncodedSMS(messageString);
+    }
+
+    // given a text message, parse out the cruft and return
+    private static String parseMessage(String newMessage){
```

Comment:
this method is no longer needed since now the url only returns the base64 encoded message. we should remove any calls made to this method